packages/web-client/app/utils/wc-provider.ts

```diff
@@ -517,8 +519,39 @@ class WalletConnectProvider extends ExtendedProviderEngine {
     };
   }
 
+  bindSocketListeners() {
+    this.websocketProvider.on('close', this.onWebsocketClose.bind(this));
+    this.websocketProvider.on('connect', this.onWebsocketConnect.bind(this));
+  }
+
+  async maybeReconnect() {
+    // setTimeout is needed to delay execution of this code until the
+    // websocket provider has finished the callback for its close event,
+    // because part of that callback includes clearing all event listeners attached to it.
+    // We need to reattach the event listeners after that completes.
+    setTimeout(() => {
+      try {
+        console.log('attempting websocket reconnection');
+        if (Date.now() - this.lastReconnection < MIN_RECONNECTION_INTERVAL) {
+          this.emit('websocket-disconnected');
+          return;
+        }
+        this.lastReconnection = Date.now();
+        this.websocketProvider.reset();
+        this.bindSocketListeners();
+        this.websocketProvider.connect();
+      } catch (e) {
+        this.emit('websocket-disconnected');
+      }
+    }, 0);
```

Comment:
Maybe this is passé but it used to be that you’d see warnings when calling Ember-y things in a `setTimeout` callback, they’d need to be wrapped in `Ember.run` to be part of the runloop. `Ember.next` is a convenience function for this exact case: run with no delay but within a runloop.